src/runtime.js

```diff
@@ -282,11 +288,13 @@ function preloadDeps(extensions, fnOrStruct) {
   // resolved first before executing this current extension.
   if (Array.isArray(fnOrStruct.i)) {
     const promises = fnOrStruct.i.map((dep) => {
```

Comment:
Do any extensions actually use `i` field? I thought it was just for single-pass build.

Reply:
No idea. Just saw it there and had to follow through. Should we remove it entirely?

Reply:
Yah, I can't find any caller of `build-system/compile/compile-wrappers.js` that passes an `intermediateDeps` array, so this is completely unused. We can follow up in a separate PR if you'd prefer.

Reply:
Ok. Reverted this part and I will send a separate cleanup PR to remove deps protocol.

Reply:
See https://github.com/ampproject/amphtml/pull/33152